webapp-django/crashstats/supersearch/tests/test_form_fields.py

```diff
@@ -154,3 +156,32 @@ def test_several_fields_illogically_datetimefield(self):
             field.clean,
             ['<2016-08-01', '<2016-08-02', '<2016-08-03']
         )
+
+    def test_boolean_field(self):
+        field = form_fields.BooleanField(required=False)
+        # If the input is None, leave it as None
+        cleaned_value = field.clean(None)
+        eq_(cleaned_value, None)
+
+        # The list of known truthy strings
+        for truthy in ('__TRUE__', 'true', 't', '1', 'Y', 'Yes'):
+            cleaned_value = field.clean(truthy)
+            eq_(cleaned_value, '__true__')
+
+        # Not truthy strings
+        for truthy in ('__False__', 'False', 'f', '0', 'n', 'No'):
```

Comment:
`None` is a special case and it's related to the fact that you can make the `BooleanField` *not* required. 
The name "truthy"  is poor. Will change. 

Also, this list of "falsy" strings is silly because it's never actually thing. I'll refactor it.